src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -119,24 +119,61 @@ def __init__(self, ws: WorkspaceClient, install_state: InstallState, verify_time
         self._install_state = install_state
         self._verify_timeout = verify_timeout
 
-    def run_workflow(self, step: str, skip_job_wait: bool = False):
+    def run_workflow(self, step: str, skip_job_wait: bool = False, max_wait: timedelta = timedelta(minutes=20)) -> int:
```

Comment:
max_wait should only be given for test runs

Reply:
That's interesting… I understand the argument, but am also wondering about the other main call location: CLI commands.

As things stand they can either not-wait, or wait. If waiting then right now they will error-out if the workflow takes longer than 20min. This wasn't a deliberate choice, but a default behaviour and I wonder if instead the CLI should wait indefinitely (or much longer). Changing the behaviour of the CLI in this respect is beyond the scope of this PR, but I imagine we have clients where the workflows can take hours and it's a completely normal situation instead of something erroneous.

Reply:
Could you log the issue to resolve this for the cli? For this PR, the behavior is preserved, therefore fine

Reply:
Done; I've created #2384 to cover updating the CLI behaviour.